packages/twenty-server/src/workspace/workspace-resolver-builder/constants/duplicate-criteria.constants.ts

```diff
@@ -0,0 +1,20 @@
+import { RecordDuplicateCriteria } from 'src/workspace/workspace-query-builder/interfaces/record.interface';
+
+export const duplicateCriteriaCollection: RecordDuplicateCriteria[] = [
```

Comment:
I agree with columnNames rather than fieldNames!

Reply:
@rostaklein Ok we'll go that way in this PR, but I'll create a ticket to use `fieldNames` instead of `columnNames` and do the compute in the factory to pre-process the `columnNames` based on the `fieldNames`.
This PR is big enough, so let's split that haha !